azurerm/resource_arm_network_watcher_flow_log.go

```diff
@@ -0,0 +1,350 @@
+	statusParameters := network.FlowLogStatusParameters{
+		TargetResourceID: &networkSecurityGroupID,
+	}
+	future, err := client.GetFlowLogStatus(ctx, resourceGroupName, networkWatcherName, statusParameters)
+	if err != nil {
+		return fmt.Errorf("Error getting Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for retrieval of Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+	}
+
+	fli, err := future.Result(client)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+	}
+
+	// There is no delete in Azure API. Disabling flow log is effectively a delete in Terraform.
+	if *fli.FlowLogProperties.Enabled {
+		fli.FlowLogProperties.Enabled = utils.Bool(false)
+
+		if isDefaultDisabledFlowLogTrafficAnalytics(fli.FlowAnalyticsConfiguration) {
+			fli.FlowAnalyticsConfiguration = nil
+		}
+
+		setFuture, err := client.SetFlowLogConfiguration(ctx, resourceGroupName, networkWatcherName, fli)
+		if err != nil {
+			return fmt.Errorf("Error disabling Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+		}
+
+		if err = setFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Error waiting for completion of disabling Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+		}
+	}
+
+	return nil
+}
+
+func expandAzureRmNetworkWatcherFlowLogRetentionPolicy(d *schema.ResourceData) *network.RetentionPolicyParameters {
+	vs := d.Get("retention_policy").([]interface{})
+	v := vs[0].(map[string]interface{})
+	enabled := v["enabled"].(bool)
+	days := v["days"].(int)
+
+	return &network.RetentionPolicyParameters{
+		Enabled: utils.Bool(enabled),
+		Days:    utils.Int32(int32(days)),
+	}
+}
+
+func flattenAzureRmNetworkWatcherFlowLogRetentionPolicy(input *network.RetentionPolicyParameters) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+	if input != nil {
+		if input.Enabled != nil {
+			result["enabled"] = *input.Enabled
+		}
+		if input.Days != nil {
+			result["days"] = *input.Days
+		}
+	}
+
+	return []interface{}{result}
+}
+
+func flattenAzureRmNetworkWatcherFlowLogTrafficAnalytics(input *network.TrafficAnalyticsProperties) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	} else if isDefaultDisabledFlowLogTrafficAnalytics(input) {
```

Comment:
_minor_ we could probably combine these e.g.

```suggestion
	if input == nil || isDefaultDisabledFlowLogTrafficAnalytics(input) {
		return []interface{}{}
```